### DISCUSSION TITLE
mariadb connection error

### DISCUSSION THREAD

**Discussion by laabubu** [usememos/memos#3104] [Q&A]
docker run  --name memos -p 5230:5230 -v /opt/memos/.memos/:/var/opt/memos neosmemo/memos --driver mariadb --dsn 'root:123456@tcp(127.0.0.1:3306)/memos_prod'
---
Server profile
version: 0.20.1
data: /var/opt/memos
dsn: root:123456@tcp(127.0.0.1:3306)/memos_prod
addr: 
port: 5230
mode: prod
driver: mariadb
frontend: true
---
2024/03/16 03:10:59 ERROR failed to create db driver !BADKEY="unknown db driver\ngithub.com/usememos/memos/store/db.NewDBDriver\n\t/backend-build/store/db/db.go:26\nmain.init.func1\n\t/backend-build/bin/memos/main.go:48\ngithub.com/spf13/cobra.(*Command).execute\n\t/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:987\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115\ngithub.com/spf13/cobra.(*Command).Execute\n\t/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039\nmain.Execute\n\t/backend-build/bin/memos/main.go:104\nmain.main\n\t/backend-build/bin/memos/main.go:194\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:271\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1695"

**Comment by lincolnthalles** [comment]
There's no `mariadb` driver in the backend. It's `mysql`.

```bash
docker run \
  --detach \
  --init \
  --name memos \
  --publish 5230:5230 \
  --volume ~/.memos/:/var/opt/memos \
  --env MEMOS_DRIVER=mysql \
  --env MEMOS_DSN=user:password@tcp(mysql_server)/memos \
  neosmemo/memos:stable
```

See [docker-compose.dev.yaml](https://github.com/usememos/memos/blob/main/scripts/docker-compose.dev.yaml) for additional insights on running different database backends.

**Reply by laabubu** [reply]
ok, thanks  ^_^
